main.py

```diff
@@ -131,6 +131,61 @@ def calculate_interest(self, rate: float) -> None:
         print("Debit accounts do not earn interest.")
 
 
+class SavingsAccount:
+    def __init__(self, account_number: str, balance: float = 0) -> None:
+        self.account_number = account_number
+        self.balance = balance
+        self.is_locked = False
+        self.transactions = []
+
+    def deposit(self, amount: float) -> None:
+        if not self.is_locked and amount > 0:
```

Comment:
czy to operacja nie powinna próbować zakładać i ściągać locka??? do przemyślenia bo sprawdzanie czy ktoś zablokował nie gwarantuje nam że możemy wykonywać operacje może inna funkcja cos zmienia